scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -1519,3 +1558,37 @@ object JavaModule {
   private[mill] def internalOrg = coursier.core.Organization("mill-internal")
   private[mill] def internalVersion = "0+mill-internal"
 }
+
+/**
+ * A module that consists solely of dependency management
+ *
+ * To be used by other modules via `JavaModule#bomModuleDeps`
+ */
+trait BomModule extends JavaModule {
```

Comment:
Can we place this in it's own file?

Reply:
I wish, but `JavaModule` and `BomModule` reference each other (`BomModule` extends `JavaModule`, and `JavaModule#bomModuleDeps` has type `Seq[BomModule]`), so that would make acyclic unhappy

Reply:
Makes sense. It's another indicator, that we should refactor `bomModuleDeps` out of `JavaModule` as part of #4338.